haystack/utils/loading.py

```diff
@@ -186,19 +186,21 @@ def indexes(self):
     def collect_indexes(self):
         indexes = []
 
-        for app_mod in haystack_get_app_modules():
+        for module, search_index in haystack_get_search_indexes():
             try:
-                search_index_module = importlib.import_module("%s.search_indexes" % app_mod.__name__)
+                search_index_module = importlib.import_module(search_index)
             except ImportError:
-                if module_has_submodule(app_mod, 'search_indexes'):
+                module_splits = search_index.split('.')
+                if (module is not None and
```

Comment:
as mentioned in app_loading, perhaps not set the first item in the tuple to None so we can simply provide an app dir in our setting.